src/v/raft/heartbeat_manager.cc

```diff
@@ -44,49 +49,58 @@ static std::vector<heartbeat_manager::node_heartbeat> requests_for_range(
             continue;
         }
 
-        auto maybe_create_follower_request = [ptr,
-                                              last_heartbeat,
-                                              &pending_beats](
-                                               const vnode& rni) mutable {
-            // special case self beat
-            // self beat is used to make sure that the protocol will make
-            // progress when there is only on node
-            if (rni == ptr->self()) {
-                pending_beats[rni.id()].emplace_back(
-                  heartbeat_metadata{ptr->meta(), rni}, 0);
-                return;
-            }
+        auto maybe_create_follower_request =
+          [ptr, last_heartbeat, &pending_beats, &reconnect_nodes](
+            const vnode& rni) mutable {
+              // special case self beat
+              // self beat is used to make sure that the protocol will make
+              // progress when there is only on node
+              if (rni == ptr->self()) {
+                  pending_beats[rni.id()].emplace_back(
+                    heartbeat_metadata{ptr->meta(), rni}, 0);
+                  return;
+              }
 
-            if (ptr->are_heartbeats_suppressed(rni)) {
-                return;
-            }
-            auto last_append_timestamp = ptr->last_append_timestamp(rni);
-
-            if (last_append_timestamp > last_heartbeat) {
-                vlog(
-                  hbeatlog.trace,
-                  "Skipping sending beat to {} gr: {} last hb {}, last append "
-                  "{}",
-                  rni,
-                  ptr->group(),
-                  last_heartbeat.time_since_epoch().count(),
-                  last_append_timestamp.time_since_epoch().count());
-                // we already sent heartbeat, skip it
-                return;
-            }
+              if (ptr->are_heartbeats_suppressed(rni)) {
+                  return;
+              }
+              auto last_append_timestamp = ptr->last_append_timestamp(rni);
 
-            auto seq_id = ptr->next_follower_sequence(rni);
-            ptr->update_suppress_heartbeats(
-              rni, seq_id, heartbeats_suppressed::yes);
-            pending_beats[rni.id()].emplace_back(
-              heartbeat_metadata{ptr->meta(), ptr->self(), rni}, seq_id);
-        };
+              if (last_append_timestamp > last_heartbeat) {
+                  vlog(
+                    hbeatlog.trace,
+                    "Skipping sending beat to {} gr: {} last hb {}, last "
+                    "append "
+                    "{}",
+                    rni,
+                    ptr->group(),
+                    last_heartbeat.time_since_epoch().count(),
+                    last_append_timestamp.time_since_epoch().count());
+                  // we already sent heartbeat, skip it
+                  return;
+              }
+
+              auto seq_id = ptr->next_follower_sequence(rni);
```

Comment:
@mmaslankaprv can you review this line.

Reply:
@mmaslankaprv you didn't comment on this.

Reply:
sorry, missed that. This line didn't change comparing to previous implementation, it was marked as change due to formatting. It is correct.